### ISSUE TITLE
[no-magic-numbers]: Ignore file patterns

### ISSUE THREAD

**Issue by henriqueleite42** [eslint/eslint#14642]
**What rule do you want to change?**

`no-magic-numbers`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Add a new optional option, that ignores the file patterns passed in an array. 

```json
// .eslintrc

{
	"no-magic-numbers": [
		"error",
		{
			"ignoreFiles": [
				"*.spec.ts",
				"src/**/tests/**/some-path/*.test.ts",
				"ignore_this_file.ts",
			]
		}
	]
}
```

**Please provide some example code that this change will affect:**

```ts
// some-test-file.spec.ts

// Without rule
expect(result.status).toBe(400);
       ^^^^ Produces warning

// With the rule
expect(result.status).toBe(400);
       ^^^^ Produces NO warning
```

**Are you willing to submit a pull request to implement this change?**

Well, I would like to try, but I need to know how (or even if it's possible) do I get the current filename that is been validated :sweat_smile:

**Comment by mdjermanovic** [comment]
Hi @henriqueleite42, thanks for the issue!

We don't have per-rule options like this, because it is possible to override settings based on glob patterns in a configuration file:

https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns

If you want that a rule ignores certain files, you can turn it `"off"` for specified patterns:

```json
{
    "rules": {
        "no-magic-numbers": "error"
    },
    "overrides":[
        {
            "files": [
                "*.spec.ts",
                "src/**/tests/**/some-path/*.test.ts",
                "ignore_this_file.ts"
            ],
            "rules": {
                "no-magic-numbers": "off"
            }
        }
    ]
}

```

I'm closing this issue because ESLint already has a feature that supports this use case.
